service/src/main/java/org/ehrbase/service/TemplateDBStorageService.java

```diff
@@ -48,14 +51,20 @@ public TemplateDBStorageService(DSLContext context, ServerConfig serverConfig) {
 
     @Override
     public List<TemplateMetaData> listAllOperationalTemplates() {
-
         return I_TemplateStoreAccess.fetchAll(getDataAccess());
     }
 
+    // Extracts and stores the template ID in a separate column?
+    @Override
+    public Set<String> findAllTemplateIds() {
+        return listAllOperationalTemplates()
+            .stream()
+            .map(metadata -> TemplateUtils.getTemplateId(metadata.getOperationaltemplate()))
+            .collect(Collectors.toSet());
+    }
```

Comment:
// Extracts and stores the template ID in a separate column?

Yes please otherwise this will be super slow.

Reply:
There is already a Collum template_id in the db